ts/a11y/explorer/KeyExplorer.ts

```diff
@@ -219,29 +221,34 @@ export class SpeechExplorer extends AbstractKeyExplorer<string> {
   public Start() {
     if (!this.attached) return;
     let options = this.getOptions();
-    // TODO: Check and set locale not only on init, but on every start.
     if (!this.init) {
       this.init = true;
-      sreReady().then(() => {
-        if (SRE.engineSetup().locale !== options.locale) {
-          SRE.setupEngine({locale: options.locale});
-        }
-        sreReady().then(() => {
-          this.Speech(this.walker);
-          this.Start();
+      SpeechExplorer.updatePromise.then(() => {
+        SpeechExplorer.updatePromise = new Promise((res) => {
+          Sre.sreReady()
+            .then(() => Sre.setupEngine({locale: options.locale}))
+            .then(() => {
+              // Important that both are in the same block so speech explorers
+              // are restarted sequentially.
+              this.Speech(this.walker);
+              this.Start();
+            })
+            .then(() => res());
```

Comment:
I'm not sure this handling of promises does what you intend it to (but I may be misjudging the intent).  If you want the `this.Speech(); this.Start()` calls to be performed in the other that the original `Start()` (and `Update()`)  methods are called, then that is not guaranteed.

To see that, consider the code:

``` js
let promise = Promise.resolve();

const delay = n => new Promise((resolve, reject) => setTimeout(resolve, n*1000));

const test = (n, message) => {
  console.log('adding promise for "' + message + '"');
  promise.then(() => {
    console.log('setting promise for "' + message + '"');
    promise = new Promise((resolve, reject) => {
      console.log('waiting for "' + message + '"');
      delay(n).then(() => {console.log(message)});
    });
  }).catch(err => console.error(err));
}

test(1, 'first call');
test(.5, 'second call');
```

Here, `promise` is your `updatePromise`, `delay()` acts as `sreReady()` (and simply does a delay to imitate the potential delay in that call), and `console.log(message)` represents `this.Speach(...); this.Start();`.  Running this produces:

```
adding promise for "first call"
adding promise for "second call"
setting promise for "first call"
waiting for "first call"
setting promise for "second call"
waiting for "second call"
second call
first call
```

Note that the second call is made before the first call.  To understand this, first note that the two `test()` calls are both made before any promises execute (the two `adding promise...` messages at the top).  That means that the same `updatePromise` is used for both, and so there are two `then()` calls made on that same promise.  When there is an idle time, the first `then()` is run, which changes `updatePromise` (the `setting promiser "first call"` message) and returns and reenters the idle state.  That allows the newly created promise to run (`waiting for "first call"`), which sets up the delay for the function that logs the first call.  Then we enter idle state again, and that allows the second promise that was attached to the original `updatePromise` from `test(.5)` to run (`setting promise for "second call"`), which then replaces the `updatePromise` with another new promise (wiping out the one created by the first `test(1)` call without ever using it, so that promise has no purpose), and again enters idle state, allowing the newly created promise to run (`waiting for "second call"`), setting up the second delay.  These two delays are independent of each other, and so when the shorter one (the second call) is satisfied after .5 seconds, its `console.log()` is performed (`second call`), and finally the other delay is reached and its message reported (`first call`).  So the order of execution of the central code of the promises is reversed. 

If you want to actions of `test(1)` and `test(.5)` to be performed sequentially, regardless of the delays), you need to update `promise` at the time `test()` is called.  Since `promise.then()` itself returns a promise, you can use that to update `promise`, as follows:

``` js
let promise = Promise.resolve();

const delay = n => new Promise((resolve, reject) => setTimeout(resolve, n*1000));

const test = (n, message) => {
  console.log('adding promise for "' + message + '"');
  promise = promise.then(() => {
    console.log('waiting for "' + message + '"');
    return delay(n).then(() => {console.log(message)});
  }).catch(err => console.error(err));
}

test(1, 'first call');
test(.5, 'second call');
```

This simplifies the process and avoids an explicit `new Promise()` that isn't really needed, while guaranteeing that the actions caused by `test()` will be performed sequentially.  Here the output is

```
adding promise for "first call"
adding promise for "second call"
waiting for "first call"
first call
waiting for "second call"
second call
```

Here, the first `test(1)` call updates `promise` (unlike in the original) using `promise.then()` so that `promise` now won't resolve until the `delay()` and `console.log()` are complete (note that the promise from `delay().then()` is returned, and so that is the promise that is used for the new value of `promise`).  Then when `test(.5)` is called, its promise is chained off the one from `test(1)` (unlike the first code above, where the promise from test(1)` is overwritten and never used).  So the second `delay()` is not performed until after the first one is completed.  That is, we don't even start waiting for the second call until after the first call is made.

So if you are trying to use `updatePromise()` to serialize the functions within the promise, then you aren't currently doing that.  I would suggest the following:

``` js
      SpeechExplorer.updatePromise = SpeechExplorer.updatePromise.then(async () => {
        return Sre.sreReady()
          .then(() => Sre.setupEngine({locale: options.locale}))
          .then(() => {
            // Important that both are in the same block so speech explorers
            // are restarted sequentially.
            this.Speech(this.walker);
            this.Start();
          });
      })
        .catch((error: Error) => console.log(error.message));
```

(I can't make a commit suggestion because of the subtracted lin in the middle, and I didn't want to give a partial code block here.)